### PR TITLE
v8 new changes

### PR DESCRIPTION
**Pull request by dimaMachina**

### CODE REVIEW

File: packages/components/style.css [hunk 1]
```diff
@@ -5,29 +5,6 @@
...
-
-:root {
-  --nextra-navbar-height: 72px;
-}
-
-html {
-  /* font from tailwindcss/base is overwritten by Nextra styles */
-  font-family: var(--font-sans), ui-sans-serif, system-ui, sans-serif, 'Apple Color Emoji',
-    'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
-}
```
Comment (dimaMachina) [inline packages/components/style.css:20]: we need to move it to guild-layout

File: .eslintrc.cjs [hunk 2]
```diff
@@ -37,18 +37,11 @@ module.exports = {
         'hive-focus',
         'hive-focus-within',
         'nextra-hamburger',
-        '@container', // Tailwind ESLint Plugin doesn't see the Container Queries classes but it does see prefixes like @sm:
+        '@container', // Tailwind ESLint Plugin doesn't see the Container Queries classes, but it does see prefixes like @sm:
```
Comment (hasparus) [inline .eslintrc.cjs:40]: do u have sth like Grammarly running in your editor? I wanna get back to writing blog posts

Comment (dimaMachina) [reply .eslintrc.cjs:40]: yeah, I have some extension in my WebStorm, if you want a name I can take a look

File: packages/components/src/components/cookies-consent.tsx [hunk 3]
```diff
@@ -1,10 +1,11 @@
 'use client';
 
-import { useState } from 'react';
+import { ComponentProps, useState } from 'react';
 import { cn } from '../cn';
 import { CallToAction } from './call-to-action';
 
-export type CookiesConsentProps = React.HTMLAttributes<HTMLElement>;
+export type CookiesConsentProps = ComponentProps<'div'>;
```
Comment (hasparus) [inline packages/components/src/components/cookies-consent.tsx:7]: I wanted to comment about `ComponentPropsWithoutRef` but I think it's gonna be obsolete soon when React starts passing refs automatically, so `ComponentProps` is good.

image

Comment (dimaMachina) [reply packages/components/src/components/cookies-consent.tsx:7]: yeah, good catch, forget to use ...wtihoutProps

File: packages/components/style.css [hunk 4]
```diff
@@ -130,14 +107,7 @@ html {
...
 /* fix for nextra navbar for HiveNavigation */
 :root {
-  --nextra-navbar-height: 90px;
-}
-
-@media (min-width: 768px) {
-  .nextra-sidebar-container,
-  .nextra-toc > .nextra-scrollbar {
-    top: var(--nextra-navbar-height);
-  }
```
Comment (dimaMachina) [inline packages/components/style.css:140]: we don't need this because nextra already have these classes related on --nextra-navbar-height

File: packages/components/src/products.tsx [hunk 5]
```diff
@@ -266,13 +266,11 @@ export const PRODUCTS_MENU_LIST: Record<string, { type: 'separator'; title: Reac
           },
         ];
       }
-      const Logo = item.logo;
       return [
         i,
         {
           type: 'page',
           href: item.href,
-          newWindow: true,
```
Comment (dimaMachina) [inline packages/components/src/products.tsx:275]: there is no more this prop in Nextra 4

File: packages/components/src/components/hive-footer/index.tsx [hunk 6]
```diff
@@ -54,22 +57,11 @@ export const HiveFooter: FC<HiveFooterProps> = ({
...
-              onClick={event => {
-                if (window.$crisp) {
-                  window.$crisp.push(['do', 'chat:open']);
-                  event.preventDefault();
-                }
-              }}
-            >
-              Contact Us
-            </a>
+            <ContactUs />
```
Comment (hasparus) [inline packages/components/src/components/hive-footer/index.tsx:60]: I feel that the `className` should be here, because it's "footer link style", and the onClick and href are reusable, so maybe something like

```suggestion
            <a
              href="hive-focus -m-2 rounded p-2 font-medium hover:text-blue-700 hover:underline dark:hover:text-blue-100"
              {...CONTACT_LINK}
            />
```

and we could spread the same `CONTACT_LINK` props elsewhere where we have the `if (window.$crisp)` bit.

File: packages/components/src/components/hive-navigation/index.tsx [hunk 7]
```diff
@@ -72,25 +82,20 @@ export function HiveNavigation({
...
-  navLinks,
+  logo = <HiveLogoLink isHive={productName === 'Hive'} />,
+  navLinks = [
+    {
+      href: productName === 'Hive' ? '/pricing' : 'https://the-guild.dev/graphql/hive/pricing',
+      children: 'Pricing',
+    },
+  ],
   developerMenu,
+  searchProps,
```
Comment (hasparus) [inline packages/components/src/components/hive-navigation/index.tsx:93]: Why do we need `searchProps`?

File: packages/components/src/components/hive-footer/index.tsx (hunk 6 above)
Comment (dimaMachina) [reply packages/components/src/components/hive-footer/index.tsx:60]: `<a>` contains onClick, passing function from server components (hive-footer is server component) to client components is not possible

File: packages/components/src/products.tsx [hunk 8]
```diff
@@ -281,7 +279,7 @@ export const PRODUCTS_MENU_LIST: Record<string, { type: 'separator'; title: Reac
                   i > 6 && 'rounded-sm bg-gray-500 text-white dark:bg-white/10',
                 )}
               >
-                <Logo className="size-4 text-[8px]" />
+                <item.logo className="size-4 text-[8px]" />
```
Comment (hasparus) [inline packages/components/src/products.tsx:282]: I remember an obscure bug where lowercase method component broke Fast Refresh. I'll try to find it.

File: packages/components/style.css [hunk 9]
```diff
@@ -130,14 +107,7 @@ html {
...
+  --nextra-navbar-height: 82px;
```
Comment (hasparus) [inline packages/components/style.css:110]: What changed by 8px? Some margin or the HiveNavigation height?

File: website/app/layout.tsx [hunk 10]
```diff
@@ -34,6 +34,17 @@ const RootLayout: FC<{
...
+      navbarProps={{
+        navLinks: [
+          {
+            href: '/docs',
+            children: 'Documentation',
+          },
+        ],
+        searchProps: {
+          placeholder: 'Search...',
+        },
```
Comment (hasparus) [inline website/app/layout.tsx:46]: scope creep, but I just realized it would be neat for it to say "Search documentation" on large screens and "Search" on smaller ones to fill the space, like with `before` pseudoelement

File: packages/components/src/components/hive-navigation/index.tsx (hunk 7 above)
Comment (hasparus) [reply packages/components/src/components/hive-navigation/index.tsx:93]: makes sense

File: packages/components/src/components/hive-footer/index.tsx (hunk 6 above)
Comment (hasparus) [reply packages/components/src/components/hive-footer/index.tsx:60]: okay, I looked up all the callsites with `window.$crisp` and refactoring would need something tricky. sorry for bikeshedding, let's ship

File: packages/components/style.css (hunk 1 above)
Comment (hasparus) [reply packages/components/style.css:20]: all right

File: packages/components/style.css (hunk 9 above)
Comment (dimaMachina) [reply packages/components/style.css:110]: Height of hive navbar is 82 or 83px with a border 
image

File: website/app/layout.tsx (hunk 10 above)
Comment (dimaMachina) [reply website/app/layout.tsx:46]: I think too tricky because the placeholder should be always `string`

File: packages/components/src/components/hive-footer/index.tsx [hunk 11]
```diff
@@ -1,41 +1,44 @@
...
-} from './icons';
+} from '../icons/index';
+import { ContactUs } from './contact-us';
 
-export interface HiveFooterProps extends IFooterExtendedProps {
+export type HiveFooterProps = {
+  className?: string;
+  resources?: ILink[];
+  logo?: ReactNode;
+  href?: string;
```
[the remark this answers is not in the document]
Comment (hasparus) [reply packages/components/src/components/hive-footer/index.tsx:22]: Makes sense. So instead of `resources: ILink[]` prop we could do sth like

```tsx
items={[
  {
    label: 'Products',
    items: [
      { children: 'Hive', href: 'https://the-guild.dev/graphql/hive' }
    ]
  },
  {
    children: 'OSS Friends',
    href: 'https://the-guild.dev/graphql/hive/oss-friends'
  }
]}
```

File: packages/components/style.css (hunk 1 above)
Comment (hasparus) [reply packages/components/style.css:20]: update: we established in Slack DMs that we don't need to add it back

1. can't use next/font from the library
2. before we use next/font, we don't know the CSS Custom Property name
3. it's enough to do the following in the website repo

```tsx
<GuildLayout
  htmlProps={{ className: "font-sans" }}
  headProps={{ 
    children: <style>{`:root { --font-sans: ${neueMontreal.style.fontFamily}; }`
  }}
/>

File: website/app/layout.tsx [hunk 12]
```diff
@@ -1,6 +1,7 @@
-import { FC } from 'react';
+import { FC, ReactNode } from 'react';
 import { getDefaultMetadata, GuildLayout } from '@theguild/components/server';
 import '@theguild/components/style.css';
+import { GitHubIcon, PaperIcon, PencilIcon } from '@theguild/components/components/icons/index';
```
Comment (dimaMachina) [inline website/app/layout.tsx:4]:
```suggestion
import { GitHubIcon, PaperIcon, PencilIcon } from '@theguild/components';
```

File: packages/components/src/components/hive-navigation/index.tsx [hunk 13]
```diff
@@ -173,9 +173,15 @@ export function HiveNavigation({
...
-        </CallToAction>
+        {isHive ? (
+          <CallToAction variant="primary" href="https://app.graphql-hive.com/" className="ml-4">
+            Sign in
+          </CallToAction>
+        ) : (
+          <CallToAction variant="primary" href="/docs" className="ml-4">
+            Docs
+          </CallToAction>
+        )}
```
Comment (dimaMachina) [inline packages/components/src/components/hive-navigation/index.tsx]: nice!

File: website/app/layout.tsx (hunk 12 above)
Comment (hasparus) [reply website/app/layout.tsx:4]: Autoimport really does weird things sometimes.
